### PR TITLE
Upstream merger

### PR DESCRIPTION
**Pull request by AlexMorgan3817**
Now you can update fork just running this workflow

### CODE REVIEW

**Comment by AlexMorgan3817** [comment]
Ready for reviews

File: .github/workflows/upstream_merge.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+on:
+  workflow_dispatch:
+    inputs:
+      Upstream:
+        description: 'Upstream to pull'
+        required: true
+        default: 'NebulaSS13/Nebula'
+      Branch:
+        description: 'Branch of upstrem to pull'
```
Comment (MistakeNot4892) [inline .github/workflows/upstream_merge.yml]: `upstream`

**Comment by MistakeNot4892** [comment]
Does this work on your fork? We had to do all kinds of silly shit to get the changelog workflow working properly due to Github repository write permissions.

File: .github/workflows/upstream_merge.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        run: |
+          git remote add upstream https://github.com/${{ github.event.inputs.Upstream }}
+          git fetch --all
+      - name: Merge
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "Upstream Merger"
+          git merge upstream/${{ github.event.inputs.Branch }}
+      - name: Push
+        uses: ad-m/github-push-action@master
```
Comment (PsiOmegaDelta) [inline .github/workflows/upstream_merge.yml:34]: Isn't the convention to use a specified release version, and then rely on Dependabot or the like to handle (most of) the update process?

File: .github/workflows/upstream_merge.yml (hunk 1 above)
Comment (AlexMorgan3817) [reply .github/workflows/upstream_merge.yml]: upsteam

**Comment by AlexMorgan3817** [comment]
> Does this work on your fork? We had to do all kinds of silly shit to get the changelog workflow working properly due to Github repository write permissions.

Yes, but have one problem, you need to manually merge if upstream changed any of .github/workflows/*

File: .github/workflows/upstream_merge.yml (hunk 2 above)
Comment (AlexMorgan3817) [reply .github/workflows/upstream_merge.yml:34]: What?

Comment (AlexMorgan3817) [reply .github/workflows/upstream_merge.yml:34]: Sooo?

Comment (PsiOmegaDelta) [reply .github/workflows/upstream_merge.yml:34]: One head over to https://github.com/ad-m/github-push-action
One find the releases page https://github.com/ad-m/github-push-action/releases
Pick the latest relevant release and use that. In this case: ad-m/github-push-action@v0.6.0

This avoids issues with hostile and/or breaking changes when/if the workflow is updated

Then one optionally setup a Dependabot workflow.
This bot automatically opens PRs when new releases of referenced workflows are released.

File: .github/workflows/upstream_merge.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+name: Merge
```
Comment (MistakeNot4892) [inline .github/workflows/upstream_merge.yml:1]: As discussed, please leave a comment here to tell downstream that this workflow can't be used with a strict merge policy (like Neb main uses). Something like: `This workflow cannot run if your fork uses review-based or CI-based branch protection (such as requiring a review before a merge), as the Github bot will not be able to bypass the protections to push directly to master.`

**Comment by afterthought2** [comment]
Feel like including this commits us to maintaining it and providing support, which is not a commitment I'm happy making. Perhaps best left downstream.

**Comment by MistakeNot4892** [comment]
After some maint discussion, the general consensus is that this isn't really needed on main Neb. The workflow is essentially an alternative way of opening a merge request, but can't run on the main repository or any repository using protected branches, so it will never be used on Neb main, and has limited use otherwise.
Thanks for opening the PR regardless.

**Pull request closed without merging**